File: 4th semester/Systems for design and implementation/lab11WebApp/src/app/pet/shared/pet.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { Pet } from './pet.model';

@Injectable({
  providedIn: 'root'
})
export class PetService {
  private url = "http://localhost:8080/api/pets"

  constructor(private http: HttpClient) { }

  getPets(): Observable<Pet[]> {
    return this.http.get<Array<Pet>>(this.url).pipe(map(res => {return res['pets']}));
  }

  getPet(id: number): Observable<Pet> {
    return this.getPets().pipe(map(Pets => Pets.find(Pet => Pet.id === id)))
  }

  savePet(Pet: Pet): Observable<Pet> {
    console.log("save Pet", Pet)

    return this.http.post<Pet>(this.url, Pet)
  }

  updatePet(Pet: Pet): Observable<Pet> {
    return this.http.put<Pet>(this.url + "/" + Pet.id, Pet)
  }

  deletePet(id: number): Observable<any> {
    return this.http.delete(this.url + "/" + id)
  }

  filterPets(name: string) {
    if (name == "") {
      alert("you shoule write a name")
    }
    return this.http.get<Array<Pet>>(this.url + "/filter/" + name).pipe(map(res => {return res['pets']}));
  }

  // sortPets() {
  //   return this.http.get<Array<Pet>>(this.url + "/sort").pipe(map(res => {return res['pets']}));
  // }
}

File: 4th semester/Systems for design and implementation/lab10/SocketsProject/common/src/main/java/Service/PetService.java
package Service;

import Model.ValidatorException;
import Model.Pet;

//import Model

import java.util.List;
import java.util.concurrent.Future;

public interface PetService {
    String ADD = "addPet";
    String DELETE = "deletePet";
    String UPDATE = "updatePet";
    String GET = "getPets";

    Future<Pet> addPet(Long id, String name, String breed, Long wid) throws ValidatorException;
    Future<Pet> deletePet(Long id);
    Future<Pet> updatePet(Long id, String name, String breed, Long wid);
    Future<List<Pet>> getPets();
}
